common/authentication/azure/auth.go

```diff
@@ -499,6 +483,23 @@ func (s EnvironmentSettings) GetEnvironment(key string) (val string, ok bool) {
 	return metadata.GetMetadataProperty(s.Metadata, MetadataKeys[key]...)
 }
 
+// Returns true if the application is running on a cloud service with Managed Identity, including: Azure App Service, Azure Functions, Azure Service Fabric, Azure Container Apps, Azure Arc, Azure Cloud Shell.
```

Comment:
You now eliminated support for services powered by IMDS but which uses the default IMDS endpoint.

Reply:
No there are 2 separate functions